### PR TITLE
Add setPedArmor function for clientside peds

### PR DESCRIPTION
**Pull request by StrixG**

### CODE REVIEW

**Comment by StrixG** [comment]
I will make one pull request with all missing clientside ped functions.

**Comment by qaisjp** [comment]
separate pull requests are preferred

**Review by qaisjp — DISMISSED** [review]
Should return false on failure. Only getters return nil on failure.

**Comment by patrikjuvonen** [comment]
Waiting for #821 to resolve so we can make changes accordingly.

**Comment by botder** [comment]
The decision in pull request #821 (**Pin down return value on failure**) is now settled.

File: Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp [hunk 1]
```diff
@@ -2121,6 +2123,25 @@ int CLuaPedDefs::SetPedMoveAnim(lua_State* luaVM)
...
+{
+    CClientPed*      pPed;
+    float            fArmor;
+    CScriptArgReader argStream(luaVM);
+    argStream.ReadUserData(pPed);
+    argStream.ReadNumber(fArmor);
+
+    if (argStream.HasErrors())
+    {
+        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
```
Comment (patrikjuvonen) [inline Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp]: The contents of this if-block should be changed to:
```cpp
return luaL_error(luaVM, argStream.GetFullErrorMessage());
```

**Review by patrikjuvonen — CHANGES_REQUESTED** [review]

**Review by patrikjuvonen — APPROVED** [review]
Thank you! This PR is ready to be merged after 1.5.7 is released.

**Pull request merged**